### PR TITLE
fix bug in `taskmodules/common/mixins.py/RelationStatisticsMixin/format_statistics()`

### PR DESCRIPTION
**Pull request by RainbowRivey**
This PR fixes a bug introduced in #92 
Error found [here](https://github.com/bayer-int/nlp-at-scale-user-dictionaries/pull/44#discussion_r2084351044)

## Changes
- src/taskmodules/common/mixins.py:
  - `to_show` is now always of type `DataFrame`
  - use `unstack()` only if `statistics` is not empty
- tests/taskmodules/common/test_mixins.py
  - add test case for empty statistics 

## Bug description:

https://github.com/ArneBinder/pie-modules/blob/00c88cae6a9ed576715503019219d7f4bce277fe/src/pie_modules/taskmodules/common/mixins.py#L259-L265
 
Variable `to_show` may have different types (`Series` or `DataFrame`) depending on check in L261, causing
error in L265: `AttributeError: 'Series' object has no attribute 'columns'`

NOTES/TODO:
- [X] add tests for mixin

### CODE REVIEW

**Review by ArneBinder — CHANGES_REQUESTED** [review]
Thanks for the fix! Yes, please implement at least one test that shows that the fix is effective. You should first execute the test without the fix (in which case it needs to fail) to show that it actually tests the issue. Then, with the fix, the test should pass.

Also, please describe in a comment why/when it may happen that `to_show.index.names > 0` (and as result, why it is sometimes a series and sometimes a dataframe) since it looks like this is not so clear.

**Comment by ArneBinder** [comment]
@RainbowRivey what's the state of this?

**Comment by RainbowRivey** [comment]
@ArneBinder 

```python
len(to_show.index.names) > 1
```
was guaranteed by type `statistics: Dict[Tuple[str, str], int]` if there is at least one entry in `statistics`.
So i changed it to more self-explaining check: 
```python 
len(to_show.index) > 0
```

Also instead of type-checking i added conversion and comments

File: src/pie_modules/taskmodules/common/mixins.py [hunk 1]
```diff
@@ -257,9 +257,13 @@ def get_statistics(self) -> Dict[Tuple[str, str], int]:
             return {}
 
     def format_statistics(self, statistics: Dict[Tuple[str, str], int]) -> str:
+        # Statistics are converted to Series and then unstacked giving us the right DataFrame we need.
```
Comment (ArneBinder) [inline src/pie_modules/taskmodules/common/mixins.py:260]: "the right DataFrame we need" - that doesn't say anything! what is the right DataFrame we need?
EDIT: after fixing, move into if case (see below)

File: src/pie_modules/taskmodules/common/mixins.py [hunk 2]
```diff
@@ -257,9 +257,13 @@ def get_statistics(self) -> Dict[Tuple[str, str], int]:
...
+        # If there were no statistics - unstack will fail, so we manually convert to DataFrame.
         to_show = pd.Series(statistics)
-        if len(to_show.index.names) > 1:
+        if len(to_show.index) > 0:
             to_show = to_show.unstack()
+        else:
+            to_show = pd.DataFrame(to_show)
```
Comment (ArneBinder) [inline src/pie_modules/taskmodules/common/mixins.py:266]:
```python
        if len(statistics) > 0:
            to_show_series = pd.Series(statistics)
            # unstack (what level?) to (get what?) (TODO)
            to_show = to_show_series.unstack()
        else:
            # If there were no statistics, create an empty dummy dataframe.
            # TODO: simplify
            to_show = pd.DataFrame(pd.Series(statistics))
```

**Review by ArneBinder — CHANGES_REQUESTED** [review]
see below

**Review by ArneBinder — APPROVED** [review]
lgtm
